### DISCUSSION TITLE
Can't get `darkMode: 'class'` to work in a styled-components + Next setup. Any tips to debug this?

### DISCUSSION THREAD

**Discussion by tannerlinsley** [ben-rogerson/twin.macro#294] [Q&A] [answered]
I've tried going into the source to do some sanity console.log's but can't even get those to show up 🤷‍♂️ This could be my lack of how macro's work, but any tips that could even potentially help me fix my own bug would be great :)

**Comment by tannerlinsley** [comment]
I think I might have solved it... maybe. But so far, using a resolve for the macros plugin fixed the situation. Not sure why though.

**Reply by tannerlinsley** [reply]
```js
module.exports = {
  presets: ['next/babel'],
  plugins: [
    require.resolve('babel-plugin-macros'),
    ['styled-components', { ssr: true }],
  ],
}
```

**Comment by ben-rogerson** [comment, marked as answer]
I've successfully tested this in a new next+sc project with the following code:

```js
import 'twin.macro'

const DarkComponent = () => (
  <div className="dark">
    <div tw="dark:(bg-black text-white)">Black bg</div>
  </div>
)

export default DarkComponent
```

What I think is happening in your case Tanner, is you're either hitting the [babel cache issue](https://github.com/kentcdodds/babel-plugin-macros#babel-cache-problem) and/or the built-in caching within next dev server.
It's been [discussed here recently](https://github.com/ben-rogerson/twin.macro/issues/284) but the fix has been to remove the `.next` cache folder and restart your dev server.

**Reply by tannerlinsley** [reply]
Interesting. I was clearing the Babel register cache with some success, but that makes more sense. Thanks for your help!
